### PR TITLE
[BEAM-1622] Rename RunnableOnService to ValidatesRunner in poms and Jenkins jobs.

### PR DESCRIPTION
**Pull request by jasonkuster**
Signed-off-by: Jason Kuster <jasonkuster@google.com>

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
R: @kennknowles

### CODE REVIEW

File: .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy [hunk 1]
```diff
@@ -18,12 +18,12 @@
...
 import common_job_properties
 
-// This job runs the suite of RunnableOnService tests against the Dataflow
+// This job runs the suite of ValidatesRunner tests against the Dataflow
 // runner.
-mavenJob('beam_PostCommit_Java_RunnableOnService_Dataflow') {
-  description('Runs the RunnableOnService suite on the Dataflow runner.')
+mavenJob('beam_PostCommit_Java_ValidatesRunner_Dataflow') {
+  description('Runs the ValidatesRunner suite on the Dataflow runner.')
+  previousNames('beam_PostCommit_Java_RunnableOnService_Dataflow')
```
Comment (kennknowles) [inline .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy]: Are these automatically transitive, or do you need to list all of them?

File: .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy [hunk 2]
```diff
@@ -37,9 +37,9 @@ mavenJob('beam_PostCommit_Java_RunnableOnService_Dataflow') {
...
   common_job_properties.enablePhraseTriggeringFromPullRequest(
     delegate,
-    'Google Cloud Dataflow Runner RunnableOnService Tests',
-    'Run Dataflow RunnableOnService')
+    'Google Cloud Dataflow Runner ValidatesRunner Tests',
+    'Run Dataflow ValidatesRunner')
 
   // Maven goals for this job.
-  goals('-B -e clean verify -am -pl runners/google-cloud-dataflow-java -DforkCount=0 -DrunnableOnServicePipelineOptions=\'[ "--runner=org.apache.beam.runners.dataflow.testing.TestDataflowRunner", "--project=apache-beam-testing", "--tempRoot=gs://temp-storage-for-runnable-on-service-tests/" ]\'')
+  goals('-B -e clean verify -am -pl runners/google-cloud-dataflow-java -DforkCount=0 -DvalidatesRunnerPipelineOptions=\'[ "--runner=org.apache.beam.runners.dataflow.testing.TestDataflowRunner", "--project=apache-beam-testing", "--tempRoot=gs://temp-storage-for-validates-runner-tests/" ]\'')
```
Comment (kennknowles) [inline .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy]: (assuming this bucket it set up)

**Comment by kennknowles** [comment]
Mostly LGTM. The seed job changes here will need tight coupling with the poms. Probably not worth trying to keep things 100% blue, especially given our current half-blue state anyhow.So to make sure I understand - merge the PR and then run the seed job ASAP?

**Comment by kennknowles** [comment]
I'd recommend a more descriptive and context-free commit subject (and PR title). For example: `Rename RunnableOnService to ValidatesRunner in poms and Jenkins jobs`

File: .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy (hunk 1 above)
Comment (jasonkuster) [inline .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy]: It's just a rename, essentially, so erasing previous history is fine.

File: .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy (hunk 2 above)
Comment (jasonkuster) [inline .jenkins/job_beam_PostCommit_Java_ValidatesRunner_Dataflow.groovy]: Yup.

**Comment by jasonkuster** [comment]
Merge -> Seed is correct -- should be fine assuming no jobs start in the interim.

Changed PR title.

**Comment by kennknowles** [comment]
OK LGTM. I'll merge and seed.

**Pull request closed without merging**